client/app/bundles/course/assessment/submissions/components/misc/SubmissionFilter.tsx

```diff
@@ -0,0 +1,229 @@
+        <>
+          <h3 style={{ marginTop: 0, marginBottom: 0 }}>
+            {intl.formatMessage(translations.filterHeader)}
+          </h3>
+          <Autocomplete
+            key={`${tabCategories[categoryNum].id}-${tabCategories[categoryNum].title}-assesment-selector`}
+            disablePortal
+            clearOnEscape
+            options={filter.assessments}
+            getOptionLabel={(option): string => option.title}
+            renderInput={(params): React.ReactNode => {
+              return (
+                <TextField
+                  {...params}
+                  label={`${intl.formatMessage(
+                    translations.filterAssessmentLabel,
+                  )} ${tabCategories[categoryNum].title}`}
+                />
+              );
+            }}
+            onChange={(
+              _event: React.SyntheticEvent,
+              value: { id: number; title: string } | null,
+            ): void => {
+              setselectedAssessment(value);
+            }}
+          />
+          <Autocomplete
+            key={`${tabCategories[categoryNum].id}-${tabCategories[categoryNum].title}-group-selector`}
+            disablePortal
+            clearOnEscape
+            options={filter.groups}
+            getOptionLabel={(option): string => option.name}
+            renderInput={(params): React.ReactNode => {
+              return (
+                <TextField
+                  {...params}
+                  label={`${intl.formatMessage(
+                    translations.filterAssessmentLabel,
+                  )} Group`}
+                />
+              );
+            }}
+            onChange={(
+              _event: React.SyntheticEvent,
+              value: { id: number; name: string } | null,
+            ): void => {
+              setselectedGroup(value);
+            }}
+          />
+          <Autocomplete
+            key={`${tabCategories[categoryNum].id}-${tabCategories[categoryNum].title}-user-selector`}
+            disablePortal
+            clearOnEscape
+            options={filter.users}
+            getOptionLabel={(option): string => option.name}
+            renderInput={(params): React.ReactNode => {
+              return (
+                <TextField
+                  {...params}
+                  label={`${intl.formatMessage(
+                    translations.filterAssessmentLabel,
+                  )} User`}
+                />
+              );
+            }}
+            onChange={(
+              _event: React.SyntheticEvent,
+              value: { id: number; name: string } | null,
+            ): void => {
+              setselectedUser(value);
+            }}
+          />
+          <Button
+            variant="contained"
+            sx={{ width: 130 }}
+            onClick={handleFilterOnClick}
+          >
+            {intl.formatMessage(translations.filterButton)}
+          </Button>
+        </>
+      )}
+      {count > 1 && (
+        <Pagination
```

Comment:
Do you think we should separate this out from the SubmissionFilter component and placed above the SubmissionTable component inside in the SubmissionsIndex/Index.tsx?

Reply:
The pagination is quite simple to implement and most of the complicated logic will be in onChange, which will be different for every page. I think there isn't much use in abstracting it.